extensions/agent/src/data/pickScheduleData.ts

```diff
@@ -22,10 +22,12 @@ export class PickScheduleData implements IAgentDialogData {
 
 	public async initialize() {
 		let agentService = await AgentUtils.getAgentService();
-		let result = await agentService.getJobSchedules(this.ownerUri);
-		if (result && result.success) {
-			this.schedules = result.schedules;
-		}
+		await agentService.getJobSchedules(this.ownerUri).then((result) => {
```

Comment:
Why did you refactor this? We shouldn't be using .then if possible. And why are you returning true? You don't actually use the return value so why bother returning anything?

And as a side note - always make sure to update the return type for the function if you are adding one.

Reply:
Fixes